message/messaging_service.cc

```diff
@@ -1498,6 +1500,26 @@ future<> messaging_service::send_hint_mutation(msg_addr id, clock_type::time_poi
         std::move(reply_to), shard, std::move(response_id), std::move(trace_info));
 }
 
+void messaging_service::register_hint_sync_point_create(std::function<future<utils::UUID> (std::vector<gms::inet_address> target_endpoints, clock_type::time_point mark_deadline)>&& func) {
```

Comment:
I guess it is better that the sender provides the uuid, so that 1) the create and query always send the uuid 2) storage_proxy::wait_for_hints_to_be_replayed will have the same uuid for all the endpoints. Since each repair has the a uuid, we can reuse the repair uuid (id.uuid in the code) so we can relate the hints sync operation with the repair.

Reply:
Good point, done.